include/cppkafka/header_list.h

```diff
@@ -281,18 +279,13 @@ bool HeaderList<HeaderType>::empty() const {
 template <typename HeaderType>
 typename HeaderList<HeaderType>::Iterator
 HeaderList<HeaderType>::begin() const {
-    assert(handle_);
-    if (empty()) {
-        return end();
-    }
-    return Iterator(make_non_owning(handle_.get()), 0);
+    return empty() ? end() : Iterator(make_non_owning(handle_.get()), 0);
```

Comment:
What's the reason why this can't just return `Iterator(make_non_owning(handle_.get()), 0)`? Everything should be internally checking if the handle is valid or not at this point. Same for `end` below.

Reply:
Because you can't call `make_non_owning(null)` so if your handle is null, it will assert. This is based on your preference to not adapt the HeaderList to a null/non-null handle.

Reply:
Touche. Alright, let's keep it as it is then.